tools/ld-chroma-decoder/monodecoder.cpp

```diff
@@ -47,6 +47,32 @@ const char *MonoDecoder::getPixelName() const
     return config.outputYCbCr ? "GRAY16" : "RGB48";
 }
 
+bool MonoDecoder::isOutputY4m()
+{
+    return config.outputY4m;
+}
+
+QString MonoDecoder::getHeaders() const
```

Comment:
Can getHeaders be in the Decoder base class, rather than being replicated three times? The code looks the same in all three versions...

Reply:
I started out doing that, but the base class doesn't have a Configuration struct.  I gave it a private Decoder::Configuration config, but my C++ skillz are lacking and Decoder::getHeaders() returns nonsense, despite me setting config.xxxx in the derived classes.

Reply:
Oh, yes - the base class can't have a Configuration member because it needs to be one of the subclasses. Leave it as is for now then and I'll have a think...